src/Rule/AtLeast.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Validator\Rule;
+
+use Yiisoft\Validator\Rule;
+use Yiisoft\Validator\Result;
+use Yiisoft\Validator\DataSetInterface;
+
+/**
+ * AtLeastValidator checks if at least $min of many attributes are filled.
+ */
+class AtLeast extends Rule
+{
+    /**
+     * The minimum required quantity of filled attributes to pass the validation.
+     * Defaults to 1.
+     */
+    private int $min = 1;
+
+    /**
+     * The list of required attributes that will be checked.
+     */
+    private array $attributes = [];
+
+    /**
+     * Message to display in case of error
+     */
+    private string $message = 'The model is not valid. Must have at least "{min}" filled attributes.';
+
+    public function __construct(array $attributes, int $min = 1)
```

Comment:
It is better to set the min parameter via the method, if necessary.

```php
    public function min(int $value): self
    {
        $new = clone $this;
        $new->min = $value;
        return $new;
    }
```

Reply:
No problem :)

I updated the PR.

Reply:
> It is better to set the min parameter via the method, if necessary.
> 
> ```
>     public function min(int $value): self
>     {
>         $new = clone $this;
>         $new->min = $value;
>         return $new;
>     }
> ```

Or setter normal why clone ?

Reply:
> > It is better to set the min parameter via the method, if necessary.
> > ```
> >     public function min(int $value): self
> >     {
> >         $new = clone $this;
> >         $new->min = $value;
> >         return $new;
> >     }
> > ```
> 
> Or setter normal why clone ?

This is how all other validators are made. It is better to stick to one style.

And to call along the chain, as specified in the documentation.

https://github.com/yiisoft/validator/blob/master/README.md

Reply:
you can have setters in stack without needing a copy of the object, for me it doesn't make sense.